com.unity.render-pipelines.universal/Documentation~/writing-shaders-urp-basic-unlit-structure.md

```diff
@@ -0,0 +1,177 @@
+
+![The shader paints the GameObject dark red](Images/shader-examples/unlit-shader-tutorial-basic-hardcoded-color.jpg)
+
+The following section introduces you to the structure of this basic shader.
+
+<a name="basic-shaderlab-structure"></a>
+
+## Basic ShaderLab structure
+
+Unity shaders are written in a Unity-specific language called [ShaderLab](https://docs.unity3d.com/Manual/SL-Shader.html). 
+
+The shader in this example has the following blocks:
+
+* [Shader](#shader)
+* [Properties](#properties)
+* [SubShader](#subshader)
+* [Pass](#pass)
+* [HLSLPROGRAM](#hlsl)
+
+<a name="shader"></a>
+
+### Shader block
+
+A ShaderLab file starts with the `Shader` declaration.
+
+```c++
+Shader "Example/URPUnlitShaderBasic"
+```
+
+The path in this declaration determines the location of the shader in the Shader menu on a Material.
+
+![location of the shader in the Shader menu on a Material](Images/shader-examples/urp-material-ui-shader-path.png)
+
+<a name="properties"></a>
+
+### Properties block
+
+The [Properties](https://docs.unity3d.com/Manual/SL-Properties.html) block contains the declarations of properties that users can set in the Inspector window on a Material.
+
+In this example, the Properties block is empty, since this shader does not expose any Material properties that a user can define. TODO:reference to Color. 
+
+### SubShader block
+
+A ShaderLab file contains one or more [SubShader](https://docs.unity3d.com/Manual/SL-SubShader.html) blocks. When rendering a mesh, Unity selects the first SubShader that is compatible with the GPU on the target device.
+
+A SubShader block contains the __Tags__ element.
+
+```
+Tags { "RenderType" = "Opaque" "RenderPipeline" = "UniversalRenderPipeline" }
+```
+
+The `RenderPipeline` tag in this Shader instructs Unity to use this SubShader block only when the project is using the Universal Render Pipeline.
+
+For more information on SubShader Tags, see [ShaderLab: SubShader Tags](https://docs.unity3d.com/Manual/SL-SubShaderTags.html).
+
+### Pass block
+
+In this example, there is one Pass block that contains the HLSL program code. For more information on Pass blocks, see [ShaderLab: Pass](https://docs.unity3d.com/Manual/SL-Pass.html).
+
+### HLSLPROGRAM block
+
+This block contains the HLSL program code.
+
+> **NOTE**: URP shaders support only the HLSL language.
+
+This block contains the `#include` declaration with the reference to the `Core.hlsl` file.
+
+```c++
+#include "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Core.hlsl"
+```
+
+The `Core.hlsl` file contains definitions of frequently used HLSL macros and functions, and also contains #include references to other HLSL files (for example, Common.hlsl, SpaceTransforms.hlsl, etc.).
+
+For example the vertex shader in the HLSL program uses the `TransformObjectToHClip` function from the `SpaceTransforms.hlsl` file. The  function transforms vertex positions from object space to homogenous space:
```

Comment:
add comma after For example (For example, the vertex shader...)
remove double space before function (The function transforms...)